compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -90,6 +90,11 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                 let local_scope = this.local_scope();
                 let (success_block, failure_block) =
                     this.in_if_then_scope(local_scope, expr_span, |this| {
+                        // Help out coverage instrumentation by injecting a dummy statement with
+                        // the original condition's span (including `!`). This fixes #115468.
+                        if this.tcx.sess.instrument_coverage() {
```

Comment:
Should this test happen inside `push_coverage_span_marker`?

Reply:
`push_coverage_span_marker` is a method on `CFG`, which doesn't have its own copy of `tcx` or `sess`.

So the only way to perform the test would be to also pass tcx/sess as an additional argument, which seems backwards to me. Better to perform the test in the outer code that already has access to the necessary information.

Having the test outside also makes it easier to see from outside that this code has no effect on the CFG in non-coverage builds.